client/src/App.js

```diff
@@ -97,30 +99,22 @@ class App extends React.Component {
             <Route
               path="/trip/:tripId/progress/:progressId"
               exact
-              render={() =>
-                !this.state.isLoggedIn ? <Redirect to="/" /> : Progress
-              }
+              render={() => (!isLoggedIn ? <Redirect to="/" /> : Progress)}
```

Comment:
I mentioned this to @ujaved931 and the plan is to remove the `Progress` model and endpoints. A visual progress bar is something React can handle pretty efficiently with local state. I’m thinking that each `Waypoint`/`Checkpoint` has a boolean value indicating if it’s completed or not. We can simply use this to generate the progress bar, rather than making a db call

Reply:
Sounds good, I think we can remove in a future change. There is a library for progress bars that is really lightweight and looks good, I'll see if I can dig it up.